tests/test_hf_api.py

```diff
@@ -288,6 +288,76 @@ def test_move_repo_invalid_repo_id(self) -> None:
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated == "auto"
+        # Test setting gated to "manual"
+        self._api.update_repo_settings(repo_id=repo_id, gated="manual")
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated == "manual"
+        # Test disabling gating
+        self._api.update_repo_settings(repo_id=repo_id, gated=False)
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated is False
+        self._api.delete_repo(repo_id=repo_id)
+
+    def test_update_dataset_repo_settings(self):
+        repo_id = self._api.create_repo(repo_id=repo_name(), repo_type=constants.REPO_TYPE_DATASET).repo_id
+        # Test setting gated to "auto"
+        self._api.update_repo_settings(repo_id=repo_id, gated="auto", repo_type=constants.REPO_TYPE_DATASET)
+        res = self._api.dataset_info(repo_id, expand="gated")
+        assert res.gated == "auto"
+        # Test setting gated to "manual"
+        self._api.update_repo_settings(repo_id=repo_id, gated="manual", repo_type=constants.REPO_TYPE_DATASET)
+        res = self._api.dataset_info(repo_id, expand="gated")
+        assert res.gated == "manual"
+        # Test disabling gating
+        self._api.update_repo_settings(repo_id=repo_id, gated=False, repo_type=constants.REPO_TYPE_DATASET)
+        res = self._api.dataset_info(repo_id, expand="gated")
+        assert res.gated is False
+        self._api.delete_repo(repo_id=repo_id, repo_type=constants.REPO_TYPE_DATASET)
+
+    def test_update_model_repo_settings(self):
+        repo_id = self._api.create_repo(repo_id=repo_name(), repo_type=constants.REPO_TYPE_MODEL).repo_id
+        # Test setting gated to "auto"
+        self._api.update_repo_settings(repo_id=repo_id, gated="auto", repo_type=constants.REPO_TYPE_MODEL)
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated == "auto"
+        # Test setting gated to "manual"
+        self._api.update_repo_settings(repo_id=repo_id, gated="manual", repo_type=constants.REPO_TYPE_MODEL)
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated == "manual"
+        # Test disabling gating
+        self._api.update_repo_settings(repo_id=repo_id, gated=False, repo_type=constants.REPO_TYPE_MODEL)
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated is False
+        self._api.delete_repo(repo_id=repo_id, repo_type=constants.REPO_TYPE_MODEL)
+
+    def test_update_space_repo_settings(self):
+        with pytest.raises(ValueError, match=r"No space_sdk provided.*"):
+            self._api.create_repo(repo_id=repo_name(), repo_type=constants.REPO_TYPE_SPACE, space_sdk=None)
+        with pytest.raises(ValueError, match=r"Invalid space_sdk.*"):
+            self._api.create_repo(repo_id=repo_name(), repo_type=constants.REPO_TYPE_SPACE, space_sdk="something")
```

Comment:
No need to test this here. Usually tests should test only 1 thing. Repo creation is being tested elsewhere so here we want to focus only on `update_repo_settings`.